File: parish360-core/src/main/java/org/parish360/core/ceremonies/dto/WitnessInfo.java
package org.parish360.core.ceremonies.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class WitnessInfo {
    private String name;
    private String relation;
    private String parish;
    private String contact;
}

File: parish360_mobile/lib/features/ceremonies/domain/entities/witness_info.dart
class WitnessInfo {
  final String? name;
  final String? relation;
  final String? parish;
  final String? contact;

  WitnessInfo({this.name, this.relation, this.parish, this.contact});

  factory WitnessInfo.fromJson(Map<String, dynamic> json) {
    return WitnessInfo(
      name: json['name'] as String?,
      relation: json['relation'] as String?,
      parish: json['parish'] as String?,
      contact: json['contact'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'relation': relation,
      'parish': parish,
      'contact': contact,
    };
  }
}
